### PR TITLE
Bump Mixin version to 0.7

### PR DESCRIPTION
**Pull request by caseif**
This PR bumps the `stable-5` Mixin version to 0.7 and resolves incompatibilities which arise as a result.

### CODE REVIEW

**Comment by Zidane** [comment]
I'm not going to review this until @Mumfrey lets me know that 0.7 is fine for general use (I'm aware it is released).

**Comment by Mumfrey** [comment]
> I'm not going to review this until @Mumfrey lets me know that 0.7 is fine for general use (I'm aware it is released).

Yeah I'd actually recommend going to 0.7 for general stability and feature improvements.

File: src/main/java/org/spongepowered/common/mixin/core/data/MixinDataHolder.java [hunk 1]
```diff
@@ -413,9 +413,4 @@ public DataContainer toContainer() {
         return Collections.emptyList();
     }
 
-    @Override
-    public DataHolder copy() {
```
Comment (gabizou) [inline src/main/java/org/spongepowered/common/mixin/core/data/MixinDataHolder.java:417]: I believed this to be fixed , why is it being removed now?

**Comment by stephan-gh** [comment]
@caseif I have backported the update to Mixin 0.6.15 for now to fix https://github.com/SpongePowered/SpongeVanilla/issues/338 which has caused conflicts in your PR. Would you mind fixing them?

I personally think we should do the update to Mixin 0.7 in bleeding first before we backport it to the stable branches eventually.

**Comment by caseif** [comment]
Should be resolved now.

File: src/main/java/org/spongepowered/common/mixin/core/data/MixinDataHolder.java (hunk 1 above)
Comment (caseif) [reply src/main/java/org/spongepowered/common/mixin/core/data/MixinDataHolder.java:417]: Mixin 0.7 claims that it clashes with lower-level implementations. The override has been migrated to classes further down in the hierarchy where necessary `MixinSpongeUser` and `MixinTileEntity`.

Comment (Mumfrey) [reply src/main/java/org/spongepowered/common/mixin/core/data/MixinDataHolder.java:417]: What are you talking about "down the hierarchy", these bridge methods conflict **in the same class**. The *correct* fix was already applied on `bleeding`, by me. Why are you using a different approach?

Comment (ryantheleach) [inline src/main/java/org/spongepowered/common/mixin/core/data/MixinDataHolder.java]: I don't see your changes in a recent commit @Mumfrey I'd appreciate if someone could narrow it down for me.

Comment (caseif) [reply src/main/java/org/spongepowered/common/mixin/core/data/MixinDataHolder.java:417]: Sorry, you're right. I was referring to moving the implementation from `MixinDataHolder` to more specific mixins, but didn't realize there was a better solution. Will fix later today.

**Comment by Zidane** [comment]
Thanks @caseif for the PR but Minecrell went ahead and took care of this among some other house cleaning.

**Pull request closed without merging**